testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/resource/param/MultiValuedParamCustomClassTest.java

```diff
@@ -0,0 +1,330 @@
+     * @tpSince RESTEasy 3.6.1
+     */
+    @Test
+    public void testMatrixParam() throws Exception {
+        ResteasyClient client = new ResteasyClientBuilder().build();
+        try {
+            Response response;
+            response = client.target(generateBaseUrl() + "/matrixParam/customConversion_list")
+                    .matrixParam("person",name1 + "," + name2 + "," + name3).request().get();
+            Assert.assertEquals(expectedResponse, response.readEntity(String.class));
+            response.close();
+
+            response = client.target(generateBaseUrl() + "/matrixParam/customConversion_arrayList")
+                    .matrixParam("person",name1 + "," + name2 + "," + name3).request().get();
+            Assert.assertEquals(expectedResponse, response.readEntity(String.class));
+            response.close();
+
+
+            response = client.target(generateBaseUrl() + "/matrixParam/customConversion_set")
+                    .matrixParam("person",name1 + "," + name2 + "," + name3).request().get();
+            Assert.assertEquals(expectedResponse, response.readEntity(String.class));
+            response.close();
+
+            response = client.target(generateBaseUrl() + "/matrixParam/customConversion_hashSet")
+                    .matrixParam("person",name1 + "," + name2 + "," + name3).request().get();
+            Assert.assertEquals(expectedResponse, response.readEntity(String.class));
+            response.close();
+
+            response = client.target(generateBaseUrl() + "/matrixParam/customConversion_sortedSet")
+                    .matrixParam("person",name1 + "," + name2 + "," + name3).request().get();
+            Assert.assertEquals(expectedResponse, response.readEntity(String.class));
+            response.close();
+
+            response = client.target(generateBaseUrl() + "/matrixParam/customConversion_treeSet")
+                    .matrixParam("person",name1 + "," + name2 + "," + name3).request().get();
+            Assert.assertEquals(expectedResponse, response.readEntity(String.class));
+            response.close();
+
+
+            response = client.target(generateBaseUrl() + "/matrixParam/customConversion_array")
+                    .matrixParam("person",name1 + "," + name2 + "," + name3).request().get();
+            Assert.assertEquals(expectedResponse, response.readEntity(String.class));
+            response.close();
+
+        } finally {
+            client.close();
+        }
+    }
+
+    /**
+     * @tpTestDetails CookieParam test
+     * @tpSince RESTEasy 3.6.1
+     */
+    @Test
+    public void testCookieParam() throws Exception {
+        ResteasyClient client = new ResteasyClientBuilder().build();
+        try {
+            Response response;
+            String cookieExpectedResponse = person1.toString();
+
+            response = client.target(generateBaseUrl() + "/cookieParam/customConversion_list")
+                    .request().cookie("person",name1).get();
```

Comment:
Why just one person?

Reply:
My mistake, of course more persons can be used here. Cookies doesn't allow to use ',', see the spec (https://tools.ietf.org/html/rfc6265), so we need to use '-' as a separator. I updated converters like [this](https://github.com/resteasy/Resteasy/pull/1658/files#diff-0c3cf7de5c965575e5a7315f449889faR15)